merino/providers/amo/backends/dynamic.py

```diff
@@ -46,6 +46,13 @@ async def _fetch_addon(
         """Fetch addon metadata via AMO. Return `None` on errors."""
         try:
             res = await client.get(f"{self.api_url}/{addon_key}", follow_redirects=True)
+        except httpx.HTTPError as e:
+            logger.error(
+                f"Addons API failed request to fetch addon: {addon_key}, {e}, {e.__class__}"
+            )
+            return None
+
```

Comment:
Great addition to determine whether it's the API request or the key!

In looking at this, I'm not sure I see the value in `e.__class__`.  We will get the class as a part of `e`, so in the case of the tests, we get:
 `TimeoutException('timedout!')` for `e`
 `<class 'httpx.TimeoutException'>` for `e.__class__`. 

Given it's kind of the same I'm not sure we need it, since the except statements will trigger from something that's a descendant of `httpx.HTTPError`, so I don't know if it would help all that much. What do you think? 

```suggestion
        except httpx.HTTPError as e:
            logger.error(
                f"Addons API failed request to fetch addon: {addon_key}, {e}"
            )
            return None

```

Reply:
Ah, so the current problem is that our errors are logging empty "e". So currently, the only thing we see in the logs are literally: `Addons API could not find key: read-aloud, ` etc. I was hoping that adding the class will let me know which subclass is causing the error to help with investigating the actual issue (as `HttpError` [is really broad](https://www.python-httpx.org/exceptions/)). The only thing that I'm pretty sure it's not is `HTTPStatusError` (not 100, but that _should_ log something... usually some sort of request code in the message).